packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
         this.lineChunks = updatedLineChunks;
         this.chunks = updatedChunks;
         return;
       }
-      let cursor = edit.getDoc().getCursor();
+      let cursor = editor.state.selection.main.head
       let newLines = splitLines(this.model.remote!);
-      let start = edit.getDoc().firstLine();
-      let last = edit.getDoc().lastLine() + 1;
-      let cumulativeOffset = 0;
-      let end: number;
-      let updatedEnd: number;
-      // We want to replace contents of editor, but if we have collapsed regions
-      // some lines have been optimized away. Carefully replace the relevant bits:
-      for (let range of this.collapsedRanges) {
-        let baseLine = range.line;
-        end = getMatchingEditLine(baseLine, this.chunks);
-        updatedEnd = getMatchingEditLine(baseLine, updatedChunks);
-        let offset = updatedEnd - end;
-        if (end !== start || offset !== 0) {
-          edit.getDoc().replaceRange(
-            newLines.slice(start + cumulativeOffset, updatedEnd + cumulativeOffset - 1).join(''),
-            CodeMirror.Pos(start, 0),
-            CodeMirror.Pos(end - 1, 0),
-            'syncModel'
-          );
-        }
-        cumulativeOffset += offset;
-        start = end + range.size;
-      }
-      if (start < last) {
-        // Only here if no collapsed ranges, replace full contents
-        edit.getDoc().replaceRange(
-          newLines.slice(start, newLines.length).join(''),
-          CodeMirror.Pos(start, 0),
-          CodeMirror.Pos(last, 0),
-          'syncModel'
-        );
-      }
-      this.ownEditor.getDoc().setCursor(cursor);
+      editor.dispatch({
+        changes: {from: 0, to: editor.state.doc.length, insert: newLines.slice(0, newLines.length).join('')}
+      });
+      this.remoteEditorWidget.cm.dispatch({selection: {anchor: cursor}})
       this.lineChunks = updatedLineChunks;
       this.chunks = updatedChunks;
-    }
   }
 
   buildGap(): HTMLElement {
-    let lock = this.lockButton = elt('div', undefined, 'CodeMirror-merge-scrolllock');
+    let lock = (this.lockButton = elt('div', undefined, 'cm-merge-scrolllock'));
     lock.title = 'Toggle locked scrolling';
-    let lockWrap = elt('div', [lock], 'CodeMirror-merge-scrolllock-wrap');
-    let self: DiffView = this;
-    CodeMirror.on(lock, 'click', function() {
-      self.setScrollLock(!self.lockScroll);
-    });
-    return this.gap = elt('div', [lockWrap], 'CodeMirror-merge-gap');
+    let lockWrap = elt('div', [lock], 'cm-merge-scrolllock-wrap');
+    lock.innerHTML = '\u21db&nbsp;&nbsp;\u21da';
+    lock.addEventListener("scroll", (event) => {
+       this.setScrollLock(!this.lockScroll);
+      });
+    return (this.gap = elt('div', [lockWrap], 'cm-merge-gap'));
   }
 
   setScrollLock(val: boolean, action?: boolean) {
     this.lockScroll = val;
     if (val && action !== false) {
-      this.syncScroll(EventDirection.OUTGOING);
+      this.syncScroll(this.baseEditorWidget.cm, this.remoteEditorWidget.cm);
```

Comment:
I don't htink we need this call here, scroll synchronization should be done by listeners registered in the init method.

Reply:
Thanks. This should have been taken into account